atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java

```diff
@@ -1904,6 +1921,31 @@ public void transactionStillCommittedEvenIfCallbackThrows() {
         });
     }
 
+    @Test
+    public void transactionStillCommittedEvenIfConditionCleanupThrows() {
+        PreCommitCondition preCommitCondition = new PreCommitCondition() {
+            @Override
+            public void throwIfConditionInvalid(long timestamp) {}
+
+            @Override
+            public void cleanup() {
+                throw new RuntimeException("boom");
+            }
+        };
+
+        assertThatThrownBy(() -> txManager.runTaskWithConditionThrowOnConflict(preCommitCondition, (txn, condition) -> {
+                    txn.put(TABLE, ImmutableMap.of(TEST_CELL, PtBytes.toBytes("tom")));
+                    return null;
+                }))
+                .isInstanceOf(RuntimeException.class)
+                .hasMessageContaining("boom");
```

Comment:
I normally prefer to have an exception declared as a constant and reference it both in your preCommitCondition and assertion (that they're equal)

Reply:
Yeah that's reasonable - I admit I copied the onSuccess version of this directly above